### ISSUE TITLE
Lua Error: attempt to concatenate field 'destName' (a nil value)

### ISSUE THREAD

**Issue by timzwiebel** [DeadlyBossMods/DBM-WoTLKC#1]
**Describe the bug**
My guild was doing the Eredar Twins in Sunwell and everyone was complaining about lua errors from DBM.

**Do you have an error log of what happened?**
Two bug reports from Swatter are below (they occur on different line numbers of EredarTwins.lua):

1)
Date: 2022-09-06 22:31:50
ID: -14
Error occured in: Global
Count: 5
Message: ..\AddOns\DBM-Sunwell\EredarTwins.lua line 87:
   attempt to concatenate field 'destName' (a nil value)
Debug:
   [string "@DBM-Sunwell\EredarTwins.lua"]:87: handler()
   [string "@DBM-Core\DBM-Core.lua"]:927:
      DBM-Core\DBM-Core.lua:914
   [string "=(tail call)"]: ?
   [string "@DBM-Core\DBM-Core.lua"]:927:
      DBM-Core\DBM-Core.lua:914
Locals:
self = <table> {
 stats = <table> {
 }
 multiMobPullDetection = <table> {
 }
 ignoreBestkill = false
 modId = "DBM-Sunwell"
 engagedDiffText = "25 Player - "
 revision = 20220829192444
 creatureId = 25165
 iconRestore = <table> {
 }
 OnCombatStart = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:48
 engagedDiffIndex = 176
 encounterId = 727
 localization = <table> {
 }
 announces = <table> {
 }
 optionCategories = <table> {
 }
 SPELL_AURA_APPLIED_DOSE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SPELL_CAST_START = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:61
 specwarns = <table> {
 }
 OnCombatEnd = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:55
 modelId = 23334
 lastAntiSpamRedacted2 = 982836.259000
 inCombat = true
 engagedDiff = "normal25"
 inCombatOnlyEventsRegistered = 1
 groupSpells = <table> {
 }
 CHAT_MSG_RAID_BOSS_EMOTE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:134
 SPELL_MISSED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:126
 SPELL_DAMAGE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:119
 zones = <table> {
 }
 combatInfo = <table> {
 }
 Options = <table> {
 }
 timers = <table> {
 }
 addon = <table> {
 }
 categorySort = <table> {
 }
 SPELL_AURA_APPLIED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SyncThreshold = 8
 inCombatOnlyEvents = <table> {
 }
 groupOptions = <table> {
 }
 SPELL_CAST_SUCCESS = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:75
 id = "Twins"
 usedIcons = <table> {
 }
 DefaultOptions = <table> {
 }
 vb = <table> {
 }
 multiEncounterPullDetection = <table> {
 }
}
args = <table> {
 sourceName = "Grand Warlock Alythess"
 destFlags = -2147483648
 event = "SPELL_CAST_SUCCESS"
 spellId = 45342
 sourceRaidFlags = 64
 spellName = "Conflagration"
 destRaidFlags = -2147483648
 destGUID = ""
 timestamp = 1662528709.637000
 sourceFlags = 133704
 sourceGUID = "Creature-0-4379-580-20555-25166-0000182AF3"
}
(*temporary) = <function> defined @Interface\AddOns\DBM-Core\DBM-Core.lua:6428
(*temporary) = <table> {
 stats = <table> {
 }
 multiMobPullDetection = <table> {
 }
 ignoreBestkill = false
 modId = "DBM-Sunwell"
 engagedDiffText = "25 Player - "
 revision = 20220829192444
 creatureId = 25165
 iconRestore = <table> {
 }
 OnCombatStart = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:48
 engagedDiffIndex = 176
 encounterId = 727
 localization = <table> {
 }
 announces = <table> {
 }
 optionCategories = <table> {
 }
 SPELL_AURA_APPLIED_DOSE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SPELL_CAST_START = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:61
 specwarns = <table> {
 }
 OnCombatEnd = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:55
 modelId = 23334
 lastAntiSpamRedacted2 = 982836.259000
 inCombat = true
 engagedDiff = "normal25"
 inCombatOnlyEventsRegistered = 1
 groupSpells = <table> {
 }
 CHAT_MSG_RAID_BOSS_EMOTE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:134
 SPELL_MISSED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:126
 SPELL_DAMAGE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:119
 zones = <table> {
 }
 combatInfo = <table> {
 }
 Options = <table> {
 }
 timers = <table> {
 }
 addon = <table> {
 }
 categorySort = <table> {
 }
 SPELL_AURA_APPLIED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SyncThreshold = 8
 inCombatOnlyEvents = <table> {
 }
 groupOptions = <table> {
 }
 SPELL_CAST_SUCCESS = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:75
 id = "Twins"
 usedIcons = <table> {
 }
 DefaultOptions = <table> {
 }
 vb = <table> {
 }
 multiEncounterPullDetection = <table> {
 }
}
(*temporary) = 5
(*temporary) = nil
(*temporary) = 2
(*temporary) = "attempt to concatenate field 'destName' (a nil value)"
specWarnNova = <table> {
 voiceOptionId = "Voice45329"
 hasVoice = 2
 spellId = 45329
 icon = 136201
 option = "Spe
 (I think there's a limit to how many characters are shown for the "Locals")
AddOns:
  (manually edited to show only relevant ones)
  Swatter, v2.5.6773 (SwimmingSeadragon)
  DBMCore, vd87d644
  DBMStatusBarTimers, vd87d644
  DBMSunwell, v
  DBMVPVEM, v6746e8a
  WeakAuras, v4.1.2
  WeakAurasCompanion, v1.2.0
  WeakAurasOptions, v4.1.2
  BlizRuntimeLib_enUS v3.4.0.30400 <none>
  (ck=b34)

2)
Date: 2022-09-06 22:32:06
ID: -13
Error occured in: Global
Count: 4
Message: ..\AddOns\DBM-Sunwell\EredarTwins.lua line 76:
   attempt to concatenate field 'destName' (a nil value)
Debug:
   [string "@DBM-Sunwell\EredarTwins.lua"]:76: handler()
   [string "@DBM-Core\DBM-Core.lua"]:927:
      DBM-Core\DBM-Core.lua:914
   [string "=(tail call)"]: ?
   [string "@DBM-Core\DBM-Core.lua"]:927:
      DBM-Core\DBM-Core.lua:914
Locals:
self = <table> {
 stats = <table> {
 }
 multiMobPullDetection = <table> {
 }
 ignoreBestkill = false
 modId = "DBM-Sunwell"
 engagedDiffText = "25 Player - "
 revision = 20220829192444
 creatureId = 25165
 iconRestore = <table> {
 }
 OnCombatStart = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:48
 engagedDiffIndex = 176
 encounterId = 727
 localization = <table> {
 }
 announces = <table> {
 }
 optionCategories = <table> {
 }
 SPELL_AURA_APPLIED_DOSE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SPELL_CAST_START = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:61
 specwarns = <table> {
 }
 OnCombatEnd = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:55
 modelId = 23334
 SPELL_MISSED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:126
 lastAntiSpamRedacted2 = 982836.259000
 inCombat = true
 engagedDiff = "normal25"
 inCombatOnlyEventsRegistered = 1
 groupSpells = <table> {
 }
 CHAT_MSG_RAID_BOSS_EMOTE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:134
 lastAntiSpamEpc1 = 982852.927000
 SPELL_DAMAGE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:119
 zones = <table> {
 }
 combatInfo = <table> {
 }
 Options = <table> {
 }
 timers = <table> {
 }
 addon = <table> {
 }
 categorySort = <table> {
 }
 SPELL_AURA_APPLIED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SyncThreshold = 8
 inCombatOnlyEvents = <table> {
 }
 groupOptions = <table> {
 }
 SPELL_CAST_SUCCESS = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:75
 id = "Twins"
 usedIcons = <table> {
 }
 DefaultOptions = <table> {
 }
 vb = <table> {
 }
 multiEncounterPullDetection = <table> {
 }
}
args = <table> {
 sourceName = "Lady Sacrolash"
 destFlags = -2147483648
 event = "SPELL_CAST_SUCCESS"
 spellId = 45329
 sourceRaidFlags = 128
 spellName = "Shadow Nova"
 destRaidFlags = -2147483648
 destGUID = ""
 timestamp = 1662528726.214000
 sourceFlags = 2632
 sourceGUID = "Creature-0-4379-580-20555-25165-0000182AF3"
}
(*temporary) = <function> defined @Interface\AddOns\DBM-Core\DBM-Core.lua:6428
(*temporary) = <table> {
 stats = <table> {
 }
 multiMobPullDetection = <table> {
 }
 ignoreBestkill = false
 modId = "DBM-Sunwell"
 engagedDiffText = "25 Player - "
 revision = 20220829192444
 creatureId = 25165
 iconRestore = <table> {
 }
 OnCombatStart = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:48
 engagedDiffIndex = 176
 encounterId = 727
 localization = <table> {
 }
 announces = <table> {
 }
 optionCategories = <table> {
 }
 SPELL_AURA_APPLIED_DOSE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SPELL_CAST_START = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:61
 specwarns = <table> {
 }
 OnCombatEnd = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:55
 modelId = 23334
 SPELL_MISSED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:126
 lastAntiSpamRedacted2 = 982836.259000
 inCombat = true
 engagedDiff = "normal25"
 inCombatOnlyEventsRegistered = 1
 groupSpells = <table> {
 }
 CHAT_MSG_RAID_BOSS_EMOTE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:134
 lastAntiSpamEpc1 = 982852.927000
 SPELL_DAMAGE = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:119
 zones = <table> {
 }
 combatInfo = <table> {
 }
 Options = <table> {
 }
 timers = <table> {
 }
 addon = <table> {
 }
 categorySort = <table> {
 }
 SPELL_AURA_APPLIED = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:101
 SyncThreshold = 8
 inCombatOnlyEvents = <table> {
 }
 groupOptions = <table> {
 }
 SPELL_CAST_SUCCESS = <function> defined @Interface\AddOns\DBM-Sunwell\EredarTwins.lua:75
 id = "Twins"
 usedIcons = <table> {
 }
 DefaultOptions = <table> {
 }
 vb = <table> {
 }
 multiEncounterPullDetection = <table> {
 }
}
(*temporary) = 5
(*temporary) = nil
(*temporary) = 1
(*temporary) = "attempt to concatenate field 'destName' (a nil value)"
specWarnNova = <table> {
 voiceOptionId = "Voice45329"
 ha
 (I think there's a limit to how many characters are shown for the "Locals")
AddOns:
  (manually edited to show only relevant ones)
  Swatter, v2.5.6773 (SwimmingSeadragon)
  DBMCore, vd87d644
  DBMStatusBarTimers, vd87d644
  DBMSunwell, v
  DBMVPVEM, v6746e8a
  WeakAuras, v4.1.2
  WeakAurasCompanion, v1.2.0
  WeakAurasOptions, v4.1.2
  BlizRuntimeLib_enUS v3.4.0.30400 <none>
  (ck=b34)

**To Reproduce**
Not sure, but most of the guild was complaining about DBM errors.

**Screenshots**
N/A

**Did you try having DeadlyBossMods as the only enabled addon and everything else (especially something like ElvUI) disabled?**
No

**Which version of DeadlyBossMods are you using?**
DBMCore, vd87d644

**Was it working in a previous version? If yes, which was the last good one?**
I think so? I don't recall seeing these errors before.

**Additional context**
N/A

**Comment by MysticalOS** [comment]
Interesting, so blizzard removed destName from SUCCESS event, to match OG wrath. kind of funny that they let that stay existing in classic vanilla and classic TBC..

it's true that success didn't have dest name until about mists or wod, which is why in the earlier expansions it wasn't actually possible to detect targets in events that didn't have APPLIED events by using SUCCESS.

TLDR though, this is a nerf from blizzard and will affect a few older fights but fortunately nothing in wrath. I'll silence the error

**Comment by timzwiebel** [comment]
I can confirm that SPELL_CAST_SUCCESS events for "Shadow Nova" and "Conflagration" do not have destGUID/destName. Unfortunately, I don't have combat logs from before wrath pre-patch to compare to. But here's the relevant events from this week in case it's useful. Although it looks like you've already fixed the error and pushed the fix, so this is all kind of academic anyway. Thanks! :)

**"in the earlier expansions it wasn't actually possible to detect targets in events that didn't have APPLIED events by using SUCCESS"**
I assume you could still use _ABSORBED/_MISSED/_DAMAGE events to determine the targets, right? Although _SUCCESS does come sooner, which is why I assume it's better to use that if possible.

1) "Conflagration"
9/6 22:31:46.610  SPELL_CAST_START,Creature-0-4379-580-20555-25166-0000182AF3,"Grand Warlock Alythess",0x20a48,0x40,0000000000000000,nil,0x80000000,0x80000000,45342,"Conflagration",0x4
9/6 22:31:50.107  SPELL_CAST_SUCCESS,Creature-0-4379-580-20555-25166-0000182AF3,"Grand Warlock Alythess",0x20a48,0x40,0000000000000000,nil,0x80000000,0x80000000,45342,"Conflagration",0x4,Creature-0-4379-580-20555-25166-0000182AF3,0000000000000000,1952125,2018275,0,0,0,-1,0,0,0,1818.21,625.04,335,4.1668,73
9/6 22:31:51.464  SPELL_AURA_APPLIED,Creature-0-4379-580-20555-25166-0000182AF3,"Grand Warlock Alythess",0x20a48,0x40,Player-XXXX-XXXXXXXX,"XXXXXXX-OldBlanchy",0x514,0x0,45342,"Conflagration",0x4,DEBUFF
9/6 22:31:52.444  (various SPELL_ABSORBED/SPELL_PERIODIC_MISSED/SPELL_PERIODIC_DAMAGE events for the player affected)
9/6 22:32:01.451  SPELL_AURA_REMOVED,Creature-0-4379-580-20555-25166-0000182AF3,"Grand Warlock Alythess",0x20a48,0x40,Player-XXXX-XXXXXXXX,"XXXXXXX-OldBlanchy",0x514,0x0,45342,"Conflagration",0x4,DEBUFF

2) "Shadow Nova"
9/6 22:32:03.178  SPELL_CAST_START,Creature-0-4379-580-20555-25165-0000182AF3,"Lady Sacrolash",0xa48,0x80,0000000000000000,nil,0x80000000,0x80000000,45329,"Shadow Nova",0x20
9/6 22:32:06.685  SPELL_CAST_SUCCESS,Creature-0-4379-580-20555-25165-0000182AF3,"Lady Sacrolash",0xa48,0x80,0000000000000000,nil,0x80000000,0x80000000,45329,"Shadow Nova",0x20,Creature-0-4379-580-20555-25165-0000182AF3,0000000000000000,1214321,2018275,0,0,0,-1,0,0,0,1803.44,638.39,335,3.0109,73
9/6 22:32:06.989  (various SPELL_DAMAGE/SPELL_ABSORBED/SPELL_MISSED events for the players affected)

**Comment by MysticalOS** [comment]
I forgot to close issue, wrath release 3.4.9 should silence the errors
